### PR TITLE
Add progress bar widget

### PR DESCRIPTION
**Pull request by shkolin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/ProgressBar.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    private string $color = '';
+
+    protected function run(): string
+    {
+        $this->buildOptions();
+
+        return $this->renderProgressBar();
+    }
+
+    protected function renderProgressBar(): string
```
Comment (samdark) [inline src/ProgressBar.php:34]:
```suggestion
    private function renderProgressBar(): string
```

File: src/ProgressBar.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            $this->options['id'] = "{$this->getId()}-progressbar";
+        }
+
+        $this->options = $this->addOptions($this->options, 'progress');
+
+        if ($this->progressMax !== null) {
+            $this->options['max'] = $this->progressMax;
+        }
+
+        if ($this->progressValue > 0) {
```
Comment (samdark) [inline src/ProgressBar.php:51]: 0 is a valid value and it's different than not specifying a value. 0 is "no progress", nothing is an animation.

File: src/ProgressBar.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ * ```php
+ * echo ProgressBar::widget()->progressValue(75);
+ * ```
+ * 
+ * @link https://bulma.io/documentation/elements/progress/
+ */
+final class ProgressBar extends Widget
+{
+    private array $options = [];
+    private float $progressValue = 0;
```
Comment (samdark) [inline src/ProgressBar.php:22]: Should be `float|null`. Null should not render a value and be a default. See below.

File: tests/ProgressBarTest.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+final class ProgressBarTest extends TestCase
+{
+    public function testProgressBar(): void
+    {
+        ProgressBar::counter(0);
+
+        $html = ProgressBar::widget()->render();
+
+        $expectedHtml = <<<HTML
+<progress id="w1-progressbar" class="progress" max="100">0%</progress>
```
Comment (samdark) [inline tests/ProgressBarTest.php:18]: Value of 0 should reder the value.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ProgressBar.php (hunk 2 above)
Comment (shkolin) [reply src/ProgressBar.php]: You're right. thanks!

**Comment by samdark** [comment]
Thank you!

**Comment by shkolin** [comment]
> Thank you!

🤝

**Pull request merged**